packages/material-ui/src/useMediaQuery/useMediaQuery.js

```diff
@@ -10,18 +11,6 @@ export default function useMediaQuery(queryInput, options = {}) {
     props: {},
   });
 
-  if (process.env.NODE_ENV !== 'production') {
-    if (typeof queryInput === 'function' && theme === null) {
```

Comment:
No need for this check anymore. By using `useTheme` from core, there is default theme always.

Reply:
We had this check so developers could import useMediaQuery and keep the bundle size overhead at +1kB gzipped, not +6kB. https://github.com/mui-org/material-ui/issues/16859

Reply:
Alright, will revert this, should be fixed with the next PR for extracting common things under one package.